src/antares/craft/model/output.py

```diff
@@ -9,9 +9,96 @@
 # SPDX-License-Identifier: MPL-2.0
 #
 # This file is part of the Antares project.
+from enum import Enum
+from typing import Any
+
+import pandas as pd
+
 from pydantic import BaseModel
 
 
+class QueryFile(Enum):
+    VALUES = "values"
+    DETAILS = "details"
+    DETAILS_STSTORAGE = "details-STstorage"
+    DETAILS_RES = "details-res"
+
+
+class Frequency(Enum):
+    HOURLY = "hourly"
+    DAILY = "daily"
+    WEEKLY = "weekly"
+    MONTHLY = "monthly"
+    ANNUAL = "annual"
+
+
+class McType(Enum):
+    ALL = "mc-all"
+    IND = "mc-ind"
+
+
+class ObjectType(Enum):
+    LINKS = "links"
+    AREAS = "areas"
+
+
+class AggregationEntry(BaseModel):
+    """
+    Represents an entry for aggregation queries
+
+    Attributes:
+        query_file: The file to query.
+        frequency: "hourly", "daily", "weekly", "monthly", "annual"
+        mc_years: Monte Carlo years to include in the query. If left empty, all years are included.
+        type_ids: which links/areas to be selected (ex: "be - fr"). If empty, all are selected (comma separated)
+        columns_names: names or regexes (if query_file is of type details) to select columns (comma separated)
+    """
+
+    query_file: QueryFile
+    frequency: Frequency
+    mc_years: str = ""
+    type_ids: str = ""
+    columns_names: str = ""
+
+    def to_query(self, object_type: ObjectType) -> str:
+        mc_years = f"&mc_years={self.mc_years}" if self.mc_years else ""
+        type_ids = f"&{object_type.value}_ids={self.type_ids}" if self.type_ids else ""
+        columns_names = f"&columns_names={self.columns_names}" if self.columns_names else ""
+
+        return f"query_file={self.query_file.value}&frequency={self.frequency.value}{mc_years}{type_ids}{columns_names}&format=csv"
+
+
 class Output(BaseModel):
     name: str
     archived: bool
+
+    def __init__(self, output_service, **kwargs: Any):  # type: ignore
+        super().__init__(**kwargs)
+        self._output_service = output_service
+
+    def get_matrix(self, path: str) -> pd.DataFrame:
+        """
+        Gets the matrix of the output
+
+        Args:
+            path: output path, eg: "mc-all/areas/south/values-hourly"
+
+        Returns: Pandas DataFrame
+        """
+        full_path = f"output/{self.name}/economy/{path}"
+        return self._output_service.get_matrix(full_path)
+
+    def aggregate_values(
```

Comment:
Now that I see the code I don't know if we should keep one method with arguments or 4 distincts method. The benefit of having 4 method would be to not introduce McType and ObjectType class that I don't really like and also ensure the queryfiles can always work. I'll see with Alexander what he thinks about it

Reply:
Alexander doesn't care. So I think it's better to introduce 4 methods. You'll also have to introduce other query classes that go for every method, see the AntaresWeb code below: 
```python
class MCIndAreasQueryFile(StrEnum):
    VALUES = "values"
    DETAILS = "details"
    DETAILS_ST_STORAGE = "details-STstorage"
    DETAILS_RES = "details-res"


class MCAllAreasQueryFile(StrEnum):
    VALUES = "values"
    DETAILS = "details"
    DETAILS_ST_STORAGE = "details-STstorage"
    DETAILS_RES = "details-res"
    ID = "id"


class MCIndLinksQueryFile(StrEnum):
    VALUES = "values"


class MCAllLinksQueryFile(StrEnum):
    VALUES = "values"
    ID = "id"

```

Reply:
I think the wording FileType would be more appropriate as we'll also use it for the local implem one day

Reply:
From an user perspective these classes will probably be hard to use. We'll have to think if there's a better solution in a next PR